### PR TITLE
Initial commit interface_channel_group provider

### PR DESCRIPTION
**Pull request by chrisvanheuveln**
- moved channel_group property out of interface provider, into new interface_channel_group provider
- channel-group causes problems in nxos when it is present on an interface; e.g. you cannot change switchport modes, etc
- moving it into its own provider allows the manifest to explicitly remove it while allowing new (non-channel-group) properties to be added to the interface

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1057,6 +1052,32 @@ Enable/Disable autostate on the SVI interface. Valid values are 'true',
 Enable/Disable management on the SVI interface. Valid values are 'true', 'false', and 'default'.
 
 --
+### Type: cisco_interface
```
Comment (mikewiebe) [inline README.md]: `cisco_interface_channel_group`

File: tests/beaker_tests/cisco_interface_channel_group/test_interface_channel_group.rb [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+# tests[id][:manifest_props] - This is essentially a master list of properties
+#   that permits re-use of the properties for both :present and :absent testing
+#   without destroying the list
+# tests[id][:resource_props] - This is essentially a master hash of properties
+#   that permits re-use of the properties for both :present and :absent testing
+#   without destroying the hash
+# tests[id][:title_pattern] - (Optional) defines the manifest title.
+#
+intf = 'ethernet1/1'
+tests['default_properties'] = {
```
Comment (mikewiebe) [inline tests/beaker_tests/cisco_interface_channel_group/test_interface_channel_group.rb]: Do you need any platform checks for these tests?

Comment (chrisvanheuveln) [inline tests/beaker_tests/cisco_interface_channel_group/test_interface_channel_group.rb]: I don't believe so. channel-groups have been around for a long time. Works on n3, n6, n7, n9. Haven't tried it on n5.

File: tests/beaker_tests/cisco_interface_channel_group/test_interface_channel_group.rb [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+# Wrapper for interface_channel_group specific settings prior to calling the
+# common test_harness.
+def test_harness_interface_channel_group(tests, id)
+  return unless platform_supports_test(tests, id)
+
+  tests[id][:ensure] = :present if tests[id][:ensure].nil?
+
+  # Workaround for (ioctl) facter bug on n7k ***
+  tests[id][:code] = [0, 2] if platform[/n7k/]
```
Comment (mikewiebe) [inline tests/beaker_tests/cisco_interface_channel_group/test_interface_channel_group.rb]: Not sure why you need this change?  I have the same issue on my 5|7k but did not need to make this change in my beaker tests...

File: tests/beaker_tests/cisco_interface_channel_group/test_interface_channel_group.rb (hunk 2 above)
Comment (mikewiebe) [inline tests/beaker_tests/cisco_interface_channel_group/test_interface_channel_group.rb]: ok

File: tests/beaker_tests/cisco_interface_channel_group/test_interface_channel_group.rb (hunk 3 above)
Comment (chrisvanheuveln) [inline tests/beaker_tests/cisco_interface_channel_group/test_interface_channel_group.rb]: I was hitting a problem a problem in test_idempotence earlier which I attributed to this bug but I just reran it with the default code setting and it worked fine. Oh well, I'll remove it.

**Comment by mikewiebe** [comment]
:+1:

**Pull request merged**